clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1427,7 +1427,7 @@ private void acquire() {
         ensureNotClosed();
         long threadId = Thread.currentThread().getId();
         if (threadId != currentThread.get() && !currentThread.compareAndSet(NO_CURRENT_THREAD, threadId))
-            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access");
+            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access. Request accessing thread is " + threadId + " and it is already being accessed by " + currentThread.get());
```

Comment:
@ijuma Yes printing  thread name is not easy task, as for that we need to have a variable to store thread name who is already  accessing the resource. And we need to update it, every time. 
I think with thread Id's also the problem can be solved. As thread id's are unique and we can see by thread Id's which is causing the issue.